fdbclient/BlobCipher.cpp

```diff
@@ -49,6 +57,152 @@
 #endif
 
 #define BLOB_CIPHER_DEBUG false
+#define BLOB_CIPHER_SERIALIZATION_CHECKS false
+
+// BlobCipherEncryptHeaderRef
+
+uint32_t BlobCipherEncryptHeaderRef::getHeaderSize(const int flagVersion,
+                                                   const int authAlgoVersion,
+                                                   const EncryptCipherMode cipherMode,
+                                                   const EncryptAuthTokenMode authMode,
+                                                   const EncryptAuthTokenAlgo authAlgo) {
+	uint32_t total = sizeof(uint16_t) * 2; // sizeof(flagVersion + algoHeaderVersion)
+	if (flagVersion != 1) {
+		throw not_implemented();
+	}
+
+	total += sizeof(BlobCipherEncryptHeaderFlagsV1);
```

Comment:
optional: we could have some unit test code to assert the serialized size of BlobCipherEncryptHeaderFlagsV1 equals to sizeof(BlobCipherEncryptHeaderFlagsV1).

Reply:
I added explicit tests for header (de)serialization to the existing unit test suite: 
```
testConfigurableEncryptionHeaderNoAuthMode(minDomainId);
testConfigurableEncryptionHeaderSingleAuthMode<AUTH_TOKEN_HMAC_SHA_SIZE>(minDomainId);
testConfigurableEncryptionHeaderSingleAuthMode<AUTH_TOKEN_AES_CMAC_SIZE>(minDomainId);
```

Reply:
my suggestion was in these tests add `assert(serializedSize.size() == sizeof(BlobCipherEncryptHeaderFlagsV1)` and inline comment this is depend on by `getHeaderSize`.

Reply:
good idea! 
Will extend the test to include the above mentioned asserts.

Reply:
Done. Added explicit test to serialize individual structs along with parent struct and  check their sizes. Also used the same test to validate `getHeaderSize` method which wasn't tested so far. 
Thanks